### PR TITLE
Optimize iszero(sub(x, y)) to eq(x, y).

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    let a := calldataload(0)
+    let b := calldataload(0x20)
+    let x := sub(a, b)
+    if gt(x, 0) {
```
Comment (hrkrshnn) [inline test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul]:
```suggestion
    if iszero(x) {
```
Shouldn't it be this?

**Review by hrkrshnn — COMMENTED** [review]
Looks good except a question about the test.

File: test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        sstore(0, 1)
+    }
+}
+// ----
+// step: expressionSimplifier
+//
+// {
+//     let _1 := 0
+//     let a := calldataload(_1)
+//     if iszero(eq(a, calldataload(0x20))) { sstore(_1, 1) }
```
Comment (bshastry) [inline test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul]: This should probably be `if eq(a, calldataload(0x20)) { sstore(_1, 1) }`

File: test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul (hunk 1 above)
Comment (ekpyron) [reply test/libyul/yulOptimizerTests/expressionSimplifier/sub_eq.yul]: Well ``gt(x,0)`` will first become ``iszero(iszero(x))`` (explaining the additional ``iszero`` in the expectations). Not sure there's a good reason to complicate it like that, though, so I took the liberty to just change it.

**Comment by ekpyron** [comment]
I just added a Changelog entry and simplified the test as requested.

**Review by ekpyron — APPROVED** [review]
LGTM, but since I added a commit myself, maybe someone else approve as well (or just merge).

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**